### PR TITLE
Members of command must be at least 25 years old

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
You can't take a command job if you're sub-25.

## How This Contributes To The Skyrat Roleplay Experience
It doesn't really *make sense* for an 18-year-old to become the captain of the station, both from a lore and a real-life-sense perspective. There's been disagreement on this in the past, but I'm willing to give it a try anyway.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog

:cl:
config: Your character must be 25 or more to be in a command role.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -926,6 +926,10 @@ SUBSYSTEM_DEF(job)
 		JobDebug("[debug_prefix] Error: [get_job_unavailable_error_message(JOB_UNAVAILABLE_FLAVOUR)], Player: [player][add_job_to_log ? ", Job: [possible_job]" : ""]")
 		return JOB_UNAVAILABLE_FLAVOUR
 
+	if((player.client.prefs.read_preference(/datum/preference/numeric/age) < MIN_COMMAND_AGE) && (possible_job.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND))
+		JobDebug("[debug_prefix] Error: [get_job_unavailable_error_message(JOB_UNAVAILABLE_COMMAND_AGE)], Player: [player][add_job_to_log ? ", Job: [possible_job]" : ""]")
+		return JOB_UNAVAILABLE_FLAVOUR
```
Comment (Zonespace27) [inline code/controllers/subsystem/job.dm]:
```suggestion
		return JOB_UNAVAILABLE_COMMAND_AGE
```
guh

**Comment by coffeecoffeecoffee** [comment]
You should probably add an exemption to this for silicon characters as their age is kind of approximate / estimation by appearance.

**Comment by Nerev4r** [comment]
"... certain positions require certain degrees and/or a minimum amount of experience to hold, and no matter how smart or gifted a person is the number of years it takes to earn those degrees or the amount of experience required isn't going to change. And then there's the simple fact that ageism is a thing - people do discriminate against people they perceive as being too young to be competent at something. This means that even if your character was some sort of genius prodigy, the character would realistically be often treated in a patronizing and condescending manner at best, and at worst not taken as seriously as someone older." - Springhole

**Comment by YakumoChen** [comment]
we really should have the reasons for positions being closed listed instead of being nebulously "not open"

**Comment by ImHoppingMad** [comment]
really don't think we should lock out jobs due to age, if someone is in a head position they should roleplay a valid reason to be in that position, age or not, especially if a character has been around for a long time, and has shown numerous qualifications for that job, as someone who has played a character who has actually aged appropriately and worked his way up, but- thats just my opinion, if it gets merged, no more fluffy sheep H.O.P cause it wouldn't make sense for them to jump from 20 to 25 randomly.

**Comment by Zonespace27** [comment]
> we really should have the reasons for positions being closed listed instead of being nebulously "not open"

I'd love to, but it's out of scope for this one

**Comment by Quink-Fix** [comment]
I have yet to see a good argument as for how this will actually improve roleplay IC outside of loosely conforming to IRL standards when it comes to Administrative and skilled positions, while mechanically punishing young characters. I can't see how allowing young characters to take Command positions has actively hurt roleplay, either.

**Comment by Nerev4r** [comment]
on paper there's nothing really wrong with a young prodigy 18 year old engineer or paramedic or w/e, I've played those, I just disagree that it should mean the character gets access to command
being *not able to be Command* despite your character feeling strongly that they're ready and able for it **can be a character point.** it can be something you can use to generate friction and rp conflict with a head of staff or whatever.
but if you can just straight up be captain while being like 19, the conflict is gone. there's no 'oh why is our captain a damn teenager' because they're just allowed to do that. and if someone does critique them in a way that matters they can just be demoted. id rather 'we have no choice, we have to hire the 19 year old as an acting head of staff' and have the setting to back me saying 'I don't think that's right' instead of it just being smth we don't question. if you're playing an 18 year old chief engineer or whatever, it's not like you can't settle for just being an 18 year old engineer. that's impressive enough without the added 'and also I'm in charge of you'

**Comment by Nerev4r** [comment]
on paper there's nothing really wrong with a young prodigy 18 year old engineer or paramedic or w/e, I've played those, I just disagree that it should mean the character gets access to command
being *not able to be Command* despite your character feeling strongly that they're ready and able for it **can be a character point.** it can be something you can use to generate friction and rp conflict with a head of staff or whatever.
but if you can just straight up be captain while being like 19, the conflict is gone. there's no 'oh why is our captain a damn teenager' because they're just allowed to do that. and if someone does critique them in a way that matters they can just be demoted. id rather 'we have no choice, we have to hire the 19 year old as an acting head of staff' and have the setting to back me saying 'I don't think that's right' instead of it just being smth we don't question. if you're playing an 18 year old chief engineer or whatever, it's not like you can't settle for just being an 18 year old engineer. that's impressive enough without the added 'and also I'm in charge of you'

File: code/controllers/subsystem/job.dm [hunk 2]
```diff
@@ -926,6 +926,10 @@ SUBSYSTEM_DEF(job)
 		JobDebug("[debug_prefix] Error: [get_job_unavailable_error_message(JOB_UNAVAILABLE_FLAVOUR)], Player: [player][add_job_to_log ? ", Job: [possible_job]" : ""]")
 		return JOB_UNAVAILABLE_FLAVOUR
 
+	if((player.client.prefs.read_preference(/datum/preference/numeric/age) < MIN_COMMAND_AGE) && (possible_job.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND))
```
Comment (ghost) [inline code/controllers/subsystem/job.dm:1154]: Seconding the first comment, there should be exemptions for Silicons as their age is a much more open-ended thing to work with.
Wouldnt be too hard to add the check on this line, skip if species is ipc or silicon; but the choice is up to maintainers in the end

**Comment by Quink-Fix** [comment]
It would be better if people had the **option** to make this a character point instead of forcing them to do it through mechanical restriction. Also, ageism isn't really a good reason for people to be retaliating with demotions in the first place so I don't see where that's coming from.

**Comment by Zergspower** [comment]
A chief medical officer, prodigy or not, would take a decade or more of schooling + experience - 25 is a fairly generous point as well in relation to that

File: code/controllers/subsystem/job.dm (hunk 2 above)
Comment (tf-4) [inline code/controllers/subsystem/job.dm:1154]:
```suggestion
	if((player.client?.prefs.read_preference(/datum/preference/numeric/age) < MIN_COMMAND_AGE) && (possible_job.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND))
```

**Review by tf-4 — CHANGES_REQUESTED** [review]
I think it's fine without an exemption for silicons.

**Comment by SpaceLoveSs13** [comment]
This makes perfect sense IC and makes the people in charge be more responsible characterwise. However, while this solves the doubts IC, will this change truly reflect or change someone's playstyle as a player OOCly? I suppose they will stop doing silly things cause they are "young", but that doesnt necessarily mean they will make matured decisions. Regardless good pr logicwise.

**Comment by ImHoppingMad** [comment]
> 

hate to say it, but I've seen characters that were 19, 18, and 20 make more mature and responsible decisions then people who are playing characters that are 30, 40, and 50. It ain't bout the character age, its more bout the player and if they are there to actually do their job properly, and take the Head position seriously or not.

**Comment by jaber134** [comment]
Pointless thing to code in. If my char is 18 I'll still rp them as they're 18 (not that age ever comes up in rp to begin with) as I'm sure most people will. In any case I doubt this'll get merged.

**Comment by Zonespace27** [comment]
Drafted while I redo this to work with the toml system

**Comment by tf-4** [comment]
> Pointless thing to code in. If my char is 18 I'll still rp them as they're 18 (not that age ever comes up in rp to begin with) as I'm sure most people will. In any case I doubt this'll get merged.

This will most likely get merged I'll be real bro.

**Comment by HoodMilk** [comment]
All people are gonna do is set their age to 25 since it affects almost nothing in-game. This isn't going to stop people from not being mature in command, they're just going to increase the number on their character screen by 5.

**Comment by hugmeorelse** [comment]
> All people are gonna do is set their age to 25 since it affects almost nothing in-game. This isn't going to stop people from not being mature in command, they're just going to increase the number on their character screen by 5.

Then do that. Be 25.

**Comment by Quink-Fix** [comment]
> Then do that. Be 25.

Responses like this make this PR even more questionable.

**Comment by RimiNosha** [comment]
> we really should have the reasons for positions being closed listed instead of being nebulously "not open"

This would require TGUI-ifying job selection upstream, I bet. So doing that here isn't really on the table.

**Comment by Zonespace27** [comment]
To quote (and reword) what one maintainer said in response to this

""
It's been stated before being against placing in a gate to command roles that's anywhere above 21 years old, for the sake that character age is frankly irrelevant for most character in the 20-30 bracket, and I've legitimately had age come up in roleplay a whole two times in the last nearly two years I've been around. Going with the argument of education is entirely moot, because we don't enforce people to actually list out any form of education behind their character, they just have to be competent to play the role, and even then, the minimum level of competency is laughably low for the vast majority of jobs.

I'd much rather reward people for playing the same department long enough to be feeling confident about taking its lead, than to lock away about half of the server's population (I just checked, at the moment there's 44/91 living players that are under 25 years old currently) from roles that often struggle to find someone to play them, nevermind someone competent.

Players aren't going to be any more mature than they already are, no matter what age they set their character to. If you want Command to be more mature, add it to the Command Policy, it's not a matter that will be resolved well through code. This will just make people raise their character's age to 25 when they want to play Command, and lower it again when they're not (or just not bother and set it to 25), and nothing else will change at all.
""


Which I think are very fair points, and after reflection I'll be closing this PR, and will think of a better way of improving the direction and impact of command.

**Pull request closed without merging**